src/landscape.cpp

```diff
@@ -1028,7 +1030,7 @@ static bool FindSpring(TileIndex tile, void *)
  */
 static bool MakeLake(TileIndex tile, void *user_data)
 {
-	uint height_lake = *static_cast<uint *>(user_data);
+	uint height_lake = *static_cast<int *>(user_data);
```

Comment:
```suggestion
	int height_lake = *static_cast<int *>(user_data);
```

? Otherwise the change of pointer type makes no sense